.github/workflows/windows-build.yml

```diff
@@ -23,34 +28,53 @@ jobs:
           repository: ${{ inputs.repository }}
           submodules: recursive
 
-      - name: Configure build environment
+      - name: Configure MSVC
+        if: ${{ matrix.compiler == 'msvc' }}
         run: |
           copy env.vs2022.bat env.bat
+
+      - name: Configure clang
+        if: ${{ matrix.compiler == 'clang' }}
+        run: |
+          choco upgrade -y llvm
+          pip install ninja
+
+      - name: Configure build environment
+        run: |
           $git_ref_name = git describe --always
           echo "git_ref_name=$git_ref_name" >> $env:GITHUB_ENV
+          git submodule > submodule-status
 
-      - name: Cache Boost
-        id: cache-boost
+      - name: Cache Boost source
+        id: cache-boost-src
         uses: actions/cache@v3
         with:
           path: |
             ${{ env.BOOST_ROOT }}.7z
-            ${{ env.BOOST_ROOT }}\stage
-          key: ${{ runner.os }}-boost-${{ env.boost_version }}
+          key: ${{ runner.os }}-boost-${{ env.boost_version }}-${{ matrix.compiler }}-src
```

Comment:
In theory, the source cache should be shared by both compilers. But in reality they conflict.
![Screenshot from 2023-08-21 17-33-00](https://github.com/rime/librime/assets/26783539/2bf4af04-9c9e-46cf-b0de-a2ce8e890784)